### ISSUE TITLE
Results on CUHK-SYSU and MSMT17 

### ISSUE THREAD

**Issue by chenhao2345** [TPCD/LifelongReID#4]
Thanks for your code. I managed to reproduce the results for Market, Duke and CUHK03. However, I tried to add CUHK-SYSU and  MSMT17 into argument --test_dataset, which could not give results mentioned in the paper.  The test_mode argument seems to be not working in test_p_s.py, while test_continual_operation_neck.py only support market/duke/mix. Could you please share more details on how to test CUHK-SYSU and  MSMT17 datasets?

**Comment by hh23333** [comment]
@chenhao2345 hi chen, I also meet this issue, have you reproduce the results on CUHK-SYSU and MSMT17?

**Comment by chenhao2345** [comment]
@hh23333 My re-implementation results on CUHK-SYSU are higher than those in the paper. Results on MSMT17 are lower than those in the paper.

**Comment by ddup-gh** [comment]
@chenhao2345 Hello chen, I got an error "all query identities do not appear in gallery" when I trained on the dataset CUHK-SYSU. Do you have any idea why that might be? Do I need to make any changes to the file chuksysu.py besides the file structure？Thanks.

**Comment by chenhao2345** [comment]
@ddup-gh If I understand correctly, it's the problem in the testing code. CUHK-SYSU has only one camera label. But you have to set different camera labels for query and gallery sets. You should modify some code in operation/test_p_s.py.

**Comment by ddup-gh** [comment]
@chenhao2345 Thank you very much! You're absolutely right, I have solved this problem.

**Comment by whisperH** [comment]
> @hh23333 My re-implementation results on CUHK-SYSU are higher than those in the paper. Results on MSMT17 are lower than those in the paper.

hi, Chen. Did you solve the problem later?
 here is my results of two experiments (mAP/Rank1)
![image](https://user-images.githubusercontent.com/8840253/165003004-a53e33af-c70f-4f1e-a08c-bbdd7d810d14.png)

**Comment by whisperH** [comment]
> @ddup-gh If I understand correctly, it's the problem in the testing code. CUHK-SYSU has only one camera label. But you have to set different camera labels for query and gallery sets. You should modify some code in operation/test_p_s.py.

@chenhao2345 hi, Chen. Did you solve the problem later?
here is my results of two experiments (mAP/Rank1)

![image](https://user-images.githubusercontent.com/8840253/165003109-cd4c99c1-88e0-41c4-aa38-4a0a73f4e7a4.png)

**Comment by chenhao2345** [comment]
@whisperH No. I got results that corresponded to those in the paper on Market, Duke and CUHK03. But my results on CUHK-SYSU were higher than those in the paper. Results on MSMT17 were lower than those in the paper.

**Comment by whisperH** [comment]
@chenhao2345 陈博士您好，我又仔细检查了一下代码和超参数，发现我复现的结果和您的结果刚好相反，我的运行的结果显示Market1501、CUHK03比论文精度高、Duke基本持平，CUHK-SYSU和MSMT17 结果反而没有论文中提到的高，大约低了50%，我们有没有可能交换检查一下代码，或者看一下您当时的配置文件或者工程目录呢？

**Comment by llzhaoshuo** [comment]
> @chenhao2345 陈博士您好，我又仔细检查了一下代码和超参数，发现我复现的结果和您的结果刚好相反，我的运行的结果显示Market1501、CUHK03比论文精度高、Duke基本持平，CUHK-SYSU和MSMT17 结果反而没有论文中提到的高，大约低了50%，我们有没有可能交换检查一下代码，或者看一下您当时的配置文件或者工程目录呢？

您好，我想向您请教一下，cuhksysu数据集相关的.mat（Train.mat, TestG50.mat, Person.mat）文件应该从哪里获取呢？

**Comment by whisperH** [comment]
https://api.bbs.cvmart.net/articles/4753


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "TPCD/LifelongReID"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月28日(星期六) 晚上8:27
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [TPCD/LifelongReID] Results on CUHK-SYSU and MSMT17  (#4)





  
@chenhao2345 陈博士您好，我又仔细检查了一下代码和超参数，发现我复现的结果和您的结果刚好相反，我的运行的结果显示Market1501、CUHK03比论文精度高、Duke基本持平，CUHK-SYSU和MSMT17 结果反而没有论文中提到的高，大约低了50%，我们有没有可能交换检查一下代码，或者看一下您当时的配置文件或者工程目录呢？
  
您好，我想向您请教一下，cuhksysu数据集相关的.mat（Train.mat, TestG50.mat, Person.mat）文件应该从哪里获取呢？

**Comment by chenhao2345** [comment]
@whisperH 你好，这是我得到的结果。除了上面cuhk-sysu的camera问题，我好像没有改什么。如果有需要，可以发邮件交流hao.chen@inria.fr。
![image](https://user-images.githubusercontent.com/67144451/171658256-fb9dee9d-8aa0-4ea9-9d3a-b9d728a14c72.png)

**Comment by whisperH** [comment]
> @whisperH 你好，这是我得到的结果。除了上面cuhk-sysu的camera问题，我好像没有改什么。如果有需要，可以发邮件交流[hao.chen@inria.fr](mailto:hao.chen@inria.fr)。 ![image](https://user-images.githubusercontent.com/67144451/171658256-fb9dee9d-8aa0-4ea9-9d3a-b9d728a14c72.png)

谢谢陈博，我检查过作者代码，感觉还是有些问题的，
![image](https://user-images.githubusercontent.com/8840253/171660304-2b88821d-2b3f-4340-861f-763321012bde.png)

初始化过程中没有放到cuda上，forward过程才放到cuda上，这样会导致训练过程grad一直为none，具体可以[参考这里](https://github.com/pytorch/pytorch/issues/17484#issuecomment-467228553)，改完之后有一到两个点的提升，但是还是跟作者论文中的结果对不上，实在是不知道啥情况了，感觉很奇怪

**Comment by chenhao2345** [comment]
@whisperH 好的，谢谢告知。
